web/pandas/pdeps/0013-standardize-apply.md

```diff
@@ -0,0 +1,333 @@
+# PDEP-13: Make the Series.apply method operate Series-wise
+
+- Created: 24 August 2023
+- Status: Under discussion
+- Discussion: [#52140](https://github.com/pandas-dev/pandas/issues/52509)
+- Author: [Terji Petersen](https://github.com/topper-123)
+- Revision: 1
+
+## Abstract
+
+Currently, giving a input to `Series.apply` is treated differently depending on the type of the input:
+
+* if the input is a numpy `ufunc`, `series.apply(func)` is equivalent to `func(series)`, i.e. similar to `series.pipe(func)`.
+* if the input is a callable, but not a numpy `ufunc`, `series.apply(func)` is similar to `Series([func(val) for val in series], index=series.index)`, i.e. similar to `series.map(func)`
+* if the input is a list-like or dict-like, `series.apply(func)` is equivalent to `series.agg(func)` (which is subtly different than `series.apply`)
+
+In contrast, `DataFrame.apply` has a consistent behavior:
+
+* if the input is a callable, `df.apply(func)` always calls each columns in the DataFrame, so is similar to `func(col) for _, col in
+df.items()` + wrapping functionality
+* if the input is a list-like or dict-like, `df.apply` call each item in the list/dict and wraps the result as needed. So for example if the input is a list, `df.apply(func_list)` is equivalent to `[df.apply(func) for func in func_list]` + wrapping functionality
+
+This PDEP proposes that:
+
+- The current complex current behavior of `Series.apply` will be deprecated in Pandas 2.2.
+- Single callables given to the `.apply` methods of `Series` will in Pandas 3.0 always be called on the whole `Series`, so `series.apply(func)` will become similar to `func(series)`,
```

Comment:
4 months is way too short of a deprecation period for a change that will most likely affect the majority of our users

Reply:
I'm a bit surprised about this. I would expect users who upgrade major versions to run their code through the latest minor version before upgrading and if they get a warning, to fix them before upgrading. Same for libraries, that they would check and fix warnings before upgrading support for major versions.

Only implementing this in v4.0 seems way too slow IMO, if we decide this is a good idea in itself.

Reply:
@phofl are you sure about 

> There is no easy way to opt into the new behaviour.

? I thought the way to do that was by passing `by_row=True`?